### ISSUE TITLE
Questions about testing procedure in the paper

### ISSUE THREAD

**Issue by Pockern** [ssj9596/SCST#2]
Dear authors,

Thank you for your interesting work on video super-resolution. As someone new to this field, I'm learning a lot from your paper. I have two practical questions about the testing process:

1. The paper mentions applying the same degradation during testing. Does this mean we need to pre-process the entire test set offline for fair comparison?

2. When using the same degradation pipeline as training, I notice the final output is resized to 128×128. Does this imply testing is also performed on 512×512 patches, even though test video typically have higher resolutions(e.g., UDM10 at 720×1272) ?

Thank you for your time and expertise. I really appreciate any clarification you could provide!

Best regards,

**Comment by ssj9596** [comment]
Thank you for the detailed questions!
Below is the answer:
1. Do we need to pre-process the entire test set offline?

Yes. We first run the same degradation pipeline on every GT frame, save the resulting LR clips, and use those fixed LR frames for all methods.

2. Do we still have to test on 128 × 128 patches?


No. The 128 × 128 value is the training default. Open
https://github.com/ssj9596/SCST/blob/main/dataloader/params_realbasicvsr.yml#L113
and change：
  ```python
  test_size: [128, 128]
  ``` 
  Example for UDM10 (720 × 1272 HR), change to: 
  ```python
  test_size: [180, 320]
  ``` 
  Give it a quick try—if the shape mismatches, just swap the two numbers.

**Comment by Pockern** [comment]
> Thank you for the detailed questions! Below is the answer:
> 
> 1. Do we need to pre-process the entire test set offline?
> 
> Yes. We first run the same degradation pipeline on every GT frame, save the resulting LR clips, and use those fixed LR frames for all methods.
> 
> 2. Do we still have to test on 128 × 128 patches?
> 
> No. The 128 × 128 value is the training default. Open https://github.com/ssj9596/SCST/blob/main/dataloader/params_realbasicvsr.yml#L113 and change：
> 
> test_size: [128, 128]
> Example for UDM10 (720 × 1272 HR), change to:
> 
> test_size: [180, 320]
> Give it a quick try—if the shape mismatches, just swap the two numbers.

Thank you for your prompt and detailed response—it has been very helpful in clarifying my questions.

I have one follow-up question regarding the UDM10 example you mentioned: After resizing the LR frames to 180×320, should we correspondingly resize the GT frames to 720×1080 (i.e., 4× the LR dimensions) to ensure proper alignment for full-reference metrics (PSNR/SSIM/LPIPS)? Or is there another recommended approach to handle the resolution mismatch?

**Comment by ssj9596** [comment]
Choose 320 instead of 318 (1272/4 = 318) is because the SD input must be divisible by 16.
When computing the evaluation metrics, you should slightly resize the HR image to 720 × 1280.

**Comment by Pockern** [comment]
> Choose 320 instead of 318 (1272/4 = 318) is because the SD input must be divisible by 16. When computing the evaluation metrics, you should slightly resize the HR image to 720 × 1280.

Got it, thanks for the help!
